feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go

```diff
@@ -0,0 +1,371 @@
+
+	t.Logf("Adding NH %d as DecapEncap and NHG %d via gRIBI", NH1ID, NH1ID)
+	args.client.AddNH(t, NH1ID, "DecapEncap", *deviations.DefaultNetworkInstance, fluent.InstalledInRIB, &gribi.NHOptions{Src: secondaryTunnelSrcIP, Dest: secondaryTunnelDstIP, VrfName: vrfName})
+	args.client.AddNHG(t, NH1ID, map[uint64]uint64{NH1ID: 100}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB, &gribi.NHGOptions{BackupNHG: NH2ID})
+	t.Logf("Adding an IPv4Entry for %s with primary DecapEncap & backup decap via gRIBI", primaryTunnelDstIP)
+	args.client.AddIPv4(t, primaryTunnelDstIP+"/"+mask, NH1ID, vrfName, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+
+	p2 := args.dut.Port(t, "port2")
+	t.Log("Capture port2 status if Up")
+	gnmi.Await(t, args.dut, gnmi.OC().Interface(p2.Name()).OperStatus().State(), 1*time.Minute, oc.Interface_OperStatus_UP)
+	operStatus := gnmi.Get(t, args.dut, gnmi.OC().Interface(p2.Name()).OperStatus().State())
+	if want := oc.Interface_OperStatus_UP; operStatus != want {
+		t.Errorf("Get(DUT port2 oper status): got %v, want %v", operStatus, want)
+	}
+
+	t.Logf("Create flows with dst %s", primaryTunnelDstIP)
+	baselineFlow := createFlow(t, args.ate, args.top, "BaseFlow", primaryTunnelDstIP, &atePort2)
+	backupFlow := createFlow(t, args.ate, args.top, "BackupFlow", primaryTunnelDstIP, &atePort3)
+	t.Logf("Validate traffic passes through port2")
+	validateTrafficFlows(t, args.ate, args.top, baselineFlow, backupFlow)
+
+	t.Log("Shutdown Port2")
+	// Setting admin state down on the DUT interface.
+	// Setting the otg interface down has no effect on kne and is not yet supported in otg
+	dutP2 := args.dut.Port(t, "port2")
+	setDUTInterfaceWithState(t, args.dut, &dutPort2, dutP2, false)
+	defer setDUTInterfaceWithState(t, args.dut, &dutPort2, dutP2, true)
+
+	t.Log("Capture port2 status if down")
+	gnmi.Await(t, args.dut, gnmi.OC().Interface(p2.Name()).OperStatus().State(), 1*time.Minute, oc.Interface_OperStatus_DOWN)
+	operStatusAfterShut := gnmi.Get(t, args.dut, gnmi.OC().Interface(p2.Name()).OperStatus().State())
+	if want := oc.Interface_OperStatus_DOWN; operStatusAfterShut != want {
+		t.Errorf("Get(DUT port2 oper status): got %v, want %v", operStatus, want)
+	}
+
+	t.Log("Validate traffic passes through port3")
+	validateTrafficFlows(t, args.ate, args.top, backupFlow, baselineFlow)
+}
+
+// createFlow returns a flow from atePort1 to the dstPfx.
+func createFlow(t *testing.T, ate *ondatra.ATEDevice, top gosnappi.Config, name string, dstPfx string, dst *attrs.Attributes) gosnappi.Flow {
+
+	flow := gosnappi.NewFlow().SetName(name)
+	flow.Metrics().SetEnable(true)
+	flow.TxRx().Device().SetTxNames([]string{atePort1.Name + ".IPv4"}).SetRxNames([]string{dst.Name + ".IPv4"})
+	ethHeader := flow.Packet().Add().Ethernet()
+	ethHeader.Src().SetValue(atePort1.MAC)
+	outerIPHeader := flow.Packet().Add().Ipv4()
+	outerIPHeader.Src().SetValue(atePort1.IPv4)
+	outerIPHeader.Dst().Increment().SetStart(dstPfx).SetCount(1)
```

Comment:
This is just equivalent to the `SetValue(dstPfx)` right? (I see that it's similar in the ATE test as well, not sure of the intent there.)

Reply:
Will resolve outside of this PR.